lib/units/unit.cpp

```diff
@@ -25,12 +25,29 @@ std::string map_unit_string(const std::string &unit) {
          : unit == "M" || unit == "month" ? "mog"
                                           : unit;
 }
+
+inline constexpr auto count_eq_type = llnl::units::precise::custom::eq_type(
+    llnl::units::precise::count.base_units());
+inline constexpr auto radian_eq_type = llnl::units::precise::custom::eq_type(
+    llnl::units::precise::radian.base_units());
+
+bool is_special_unit(const llnl::units::precise_unit &unit) {
+  using namespace llnl::units::precise::custom;
+  const auto &base = unit.base_units();
+
+  // count and radian are flagged as equation units, so we need to catch those
+  // early.
```

Comment:
... but they are not using the flags that we are not storing in the file, are they?

Reply:
No:
```.cpp
const auto u = units::Unit("count").underlying();
std::cout << llnl::units::precise::custom::eq_type(u.base_units()) << '\n';  // 8
std::cout << u.is_equation() << '\n';  // 0
std::cout << u.is_per_unit() << '\n';  // 0
std::cout << u.has_e_flag() << '\n';  // 0
std::cout << u.has_i_flag() << '\n';  // 0
```
`eq_type` has special cases for counts and radians but the other four lines (which are the ones used for serialization) don't reflect this. Instead, `count` uses its own base.